mtdata/storage.py

```diff
@@ -226,14 +238,82 @@ class CSVBasic(Storage):
     A minimal CSV implementation that uses a `DictWriter` to write rows
     to the indicated file.
     """
+
+    @staticmethod
+    def name() -> str:
+        return 'csv'
+
     def append(self, name: str,
                data: Iterable[Row],
                dedup_facets: Iterable[str],
-               dedup_fields: Iterable[str]) -> None:
-        pass
+               dedup_fields: Iterable[str]) -> StoreResult:
+
+        existing_data = self.load(name)
+        if (existing_data == []):
+            return self.replace(name, data)
+        
+        deduped_data = self.dedup(
+            existing_data,
+            data,
+            dedup_facets,
+            dedup_fields,
+        )
+
+        with open(self.name_to_path(name), 'a') as file:
+            import csv
+            writer = None
+            for row in deduped_data:
+                if writer is None:
+                    writer = csv.DictWriter(file, fieldnames=list(row.keys()))
+                writer.writerow(row)
+
+        return StoreResult(
+            success=True,
+            message=''
+        )
 
     def load(self, name: str) -> Iterable[Row]:
-        pass
+        """
+        Convert csv rows into an array of dictionaries
+        All data types are automatically checked and converted
+        """
+        import ast
+        from itertools import islice
 
-    def replace(self, name: str, data: Iterable[Row]) -> None:
-        pass
+        cursor = []  # Placeholder for the dictionaries/documents
+        try:
```

Comment:
For posterity, this block was necessary because `csv.DictReader` doesn't correctly type values pulled from csvs. Adapted from https://stackoverflow.com/questions/11665628/read-data-from-csv-file-and-transform-from-string-to-correct-data-type-includin

Reply:
This is broken on the 2nd+ run, will fix